pytest/test_network.py

```diff
@@ -477,107 +506,112 @@ def testClusterRefreshOnOnlySingleNode(env):
 
-        conn = shardMock.GetConnection(sendHelloResponse=False)
+            conn = shardMock.GetConnection(sendHelloResponse=False)
 
-        # read RG.HELLO request
-        env.assertEqual(conn.read_request(), ['RG.HELLO'])
+            # read RG.HELLO request
+            env.assertEqual(conn.read_request(), ['RG.HELLO'])
 
 @gearsTest(skipOnCluster=True, skipCleanups=True, skipWithTLS=True)
 def testMassiveClusterSet(env):
-    with ShardMock(env) as shardMock:
-        for i in range(1000):
-            conn = shardMock.GetConnection(sendHelloResponse=False)
-            shardMock._send_cluster_set()
+    for host in _get_hosts():
+        with ShardMock(env, host) as shardMock:
+            for i in range(1000):
+                conn = shardMock.GetConnection(sendHelloResponse=False)
+                shardMock._send_cluster_set()
 
 @gearsTest(skipOnCluster=True, skipCleanups=True, skipWithTLS=True)
 def testMassiveClusterSetFromShard(env):
-    with ShardMock(env) as shardMock:
-        for i in range(1000):
-            env.cmd('RG.CLUSTERSETFROMSHARD',
-                     'NO-USED',
-                     'NO-USED',
-                     'NO-USED',
-                     'NO-USED',
-                     'NO-USED',
-                     '1',
-                     'NO-USED',
-                     '2',
-                     'NO-USED',
-                     '1',
-                     'NO-USED',
-                     '0',
-                     '8192',
-                     'NO-USED',
-                     'password@localhost:6379',
-                     'NO-USED',
-                     'NO-USED',
-                     '2',
-                     'NO-USED',
-                     '8193',
-                     '16383',
-                     'NO-USED',
-                     'password@localhost:%d' % shardMock.port
-                     )
+    for host in _get_hosts():
+            with ShardMock(env, host) as shardMock:
+                for i in range(1000):
+                    env.cmd('RG.CLUSTERSETFROMSHARD',
+                            'NO-USED',
+                            'NO-USED',
+                            'NO-USED',
+                            'NO-USED',
+                            'NO-USED',
+                            '1',
+                            'NO-USED',
+                            '2',
+                            'NO-USED',
+                            '1',
+                            'NO-USED',
+                            '0',
+                            '8192',
+                            'NO-USED',
+                            'password@%s:6379' % shardMock.host,
+                            'NO-USED',
+                            'NO-USED',
+                            '2',
+                            'NO-USED',
+                            '8193',
+                            '16383',
+                            'NO-USED',
+                            'password@%s:%d' % (shardMock.host, shardMock.port)
```

Comment:
What happens with :: and :0: as hosts, both of which are technically valid ipv6?

Reply:
The code searched for the last `:` and assume that only after comes the port so I do not see why it would be a problem.